### PR TITLE
22614 create journey part2

### PR DESCRIPTION
**Pull request by nnorbert**
Finished the Forms, Payments and Post-Submission step forms and data handling
Messaging part is still missing

### CODE REVIEW

File: apps/integrator/.env.sample [hunk 1]
```diff
@@ -15,6 +15,8 @@ LOGTO_BASE_URL=http://localhost:3001
 LOGTO_MESSAGING_E2E_APP_ID=r4i5m03dxklbq5psw3c61
 LOGTO_MESSAGING_E2E_APP_SECRET=messaging_e2e_app_local_secret
 
+PAYMENTS_BACKEND_URL=http://localhost:8001
```
Comment (alfonsograziano) [inline apps/integrator/.env.sample:18]: We will need to add this env var on Terraform as well

Comment (alfonsograziano) [reply apps/integrator/.env.sample:18]: Non-blocker: Since we have multiple references to the services, at some point we might think of creating a tiny services registry so that we don't have to add repeating env vars within multiple services

File: apps/integrator/app/[locale]/(hosted)/admin/journeys/details/[journeyId]/page.tsx [hunk 2]
```diff
@@ -1,10 +1,11 @@
-import { getTranslations } from "next-intl/server";
-import { notFound } from "next/navigation";
-import Link from "next/link";
-import { PageWrapper } from "../../../../PageWrapper";
-import advancedFormat from "dayjs/plugin/advancedFormat";
-import dayjs from "dayjs";
-import { AuthenticationFactory } from "../../../../../../../libraries/authentication-factory";
+// import { getTranslations } from "next-intl/server";
```
Comment (alfonsograziano) [inline apps/integrator/app/[locale]/(hosted)/admin/journeys/details/[journeyId]/page.tsx:1]: Just for clarity in case someone sees this: it will be implemented in a new ask

File: apps/integrator/libraries/journeyEditor/widgets/messagingWidget.tsx [hunk 3]
```diff
@@ -1,7 +1,85 @@
...
+          </thead>
+          <tbody className="govie-table__body">
+            <tr className="govie-table__row" key={this.noMessageKey}>
+              <td className="govie-table__cell govie-table__cell--vertical-centralized govie-body-s">
+                <div
+                  data-module="govie-radios"
+                  className="govie-radios govie-radios--large"
+                >
+                  <div className="govie-radios__item" key={0}>
+                    <input
```
Comment (alfonsograziano) [inline apps/integrator/libraries/journeyEditor/widgets/messagingWidget.tsx:55]: Maybe the name here is wrong?

File: apps/integrator/libraries/journeyEditor/widgets/messagingWidget.tsx [hunk 4]
```diff
@@ -1,7 +1,85 @@
...
+              </td>
+            </tr>
+          </tbody>
+        </table>
+      </div>
+    );
+  }
+
+  getStepData(data: FormData) {
+    return {};
```
Comment (alfonsograziano) [inline apps/integrator/libraries/journeyEditor/widgets/messagingWidget.tsx:83]: This still needs to be implemented, right? Shall we add a TODO?

File: apps/integrator/libraries/journeyEditor/widgets/titleWidget.tsx [hunk 5]
```diff
@@ -1,7 +1,11 @@
 import { JourneyStepWidget } from ".";
 
 export default class TitleWidget implements JourneyStepWidget {
-  getForm() {
+  async getForm() {
```
Comment (alfonsograziano) [inline apps/integrator/libraries/journeyEditor/widgets/titleWidget.tsx:4]: Is the implementation missing here?

File: apps/intg-api/src/routes/schemas/index.ts [hunk 6]
```diff
@@ -98,19 +98,20 @@ export const JourneyStepTypes = Type.Union([
 
 export const FormStepData = Type.Object({
   url: Type.String(),
-  title: Type.String(),
 });
 
 export const PaymentStepData = Type.Object({
-  url: Type.String(),
+  paymentRequestId: Type.String(),
```
Comment (alfonsograziano) [inline apps/intg-api/src/routes/schemas/index.ts:104]: Do we need to adapt the behavior of the executor as well? - I haven't checked yet if you did this change already in this PR

File: apps/intg-api/src/routes/schemas/index.ts [hunk 7]
```diff
@@ -98,19 +98,20 @@ export const JourneyStepTypes = Type.Union([
...
 
-export const MessagingStepData = Type.Object({});
+export const MessagingStepData = Type.Object({
+  templateId: Type.String(),
+});
 
 export const CompleteStepData = Type.Object({
-  label: Type.String(),
-  url: Type.String(),
+  buttonLabel: Type.String(),
```
Comment (alfonsograziano) [inline apps/intg-api/src/routes/schemas/index.ts:113]: Same here, not sure if you updated the executor, will resolve this conversation if I find the changes later

File: apps/payments-api/plugins/entities/paymentRequest/repo.ts [hunk 8]
```diff
@@ -20,6 +20,40 @@ export class PaymentRequestRepo {
     return this.pg.transact;
   }
 
+  getAllActivePaymentRequests(
+    organizationId: string,
+  ): Promise<QueryResult<PaymentRequestDO>> {
+    return this.pg.query(
+      `select pr.title,
```
Comment (alfonsograziano) [inline apps/payments-api/plugins/entities/paymentRequest/repo.ts:27]: Do we need all these properties? Maybe we can get rid of some of them and we should be able to remove the left join on the providers as well

File: apps/payments-api/plugins/entities/paymentRequest/repo.ts [hunk 9]
```diff
@@ -31,6 +65,8 @@ export class PaymentRequestRepo {
         pr.amount,
         pr.reference,
         pr.status,
+        pr.allow_amount_override as "allowAmountOverride",
```
Comment (alfonsograziano) [inline apps/payments-api/plugins/entities/paymentRequest/repo.ts:68]: Since we're using the other API, shall we revert this change since these fields are not required?

File: apps/integrator/libraries/journeyEditor/widgets/messagingWidget.tsx (hunk 4 above)
Comment (nnorbert) [reply apps/integrator/libraries/journeyEditor/widgets/messagingWidget.tsx:83]: Yes, messaging is still not implemented, this is just a placeholder. Added a comment.

File: apps/integrator/libraries/journeyEditor/widgets/titleWidget.tsx (hunk 5 above)
Comment (nnorbert) [reply apps/integrator/libraries/journeyEditor/widgets/titleWidget.tsx:4]: It is missing, I will implement it in the next PR, because this is not in use yet.

File: apps/intg-api/src/routes/schemas/index.ts (hunk 6 above)
Comment (nnorbert) [reply apps/intg-api/src/routes/schemas/index.ts:104]: Oh right, I have to build the payment URL now. Good catch, thanks!

File: apps/payments-api/plugins/entities/paymentRequest/repo.ts (hunk 9 above)
Comment (nnorbert) [reply apps/payments-api/plugins/entities/paymentRequest/repo.ts:68]: Right, I want to do this, and totally forgot about it. Good catch, thanks!

File: apps/integrator/libraries/journeyEditor/widgets/messagingWidget.tsx (hunk 3 above)
Comment (nnorbert) [reply apps/integrator/libraries/journeyEditor/widgets/messagingWidget.tsx:53]: Done

File: apps/intg-api/src/routes/schemas/index.ts (hunk 6 above)
Comment (nnorbert) [reply apps/intg-api/src/routes/schemas/index.ts:104]: Done

File: apps/intg-api/src/routes/schemas/index.ts (hunk 7 above)
Comment (nnorbert) [reply apps/intg-api/src/routes/schemas/index.ts:113]: Done

File: apps/payments-api/plugins/entities/paymentRequest/repo.ts (hunk 8 above)
Comment (nnorbert) [reply apps/payments-api/plugins/entities/paymentRequest/repo.ts:28]: Done

File: apps/payments-api/plugins/entities/paymentRequest/repo.ts (hunk 9 above)
Comment (nnorbert) [reply apps/payments-api/plugins/entities/paymentRequest/repo.ts]: Done

File: apps/integrator/.env.sample (hunk 1 above)
Comment (nnorbert) [reply apps/integrator/.env.sample:18]: PR: https://github.com/ogcio/life-events-terraform/pull/168

**Review by alfonsograziano — DISMISSED** [review]
LGTM

**Review by alfonsograziano — APPROVED** [review]
